deploy/osps/default/osp-amzn2.yaml

```diff
@@ -111,96 +99,232 @@ spec:
       cd -
 
-      {{- /* kubelet */}}
+      {{- /* kubelet */ }}
       KUBE_VERSION="${KUBE_VERSION:-{{ .KubeVersion }}}"
       kube_dir="$opt_bin/kubernetes-$KUBE_VERSION"
       kube_base_url="https://storage.googleapis.com/kubernetes-release/release/$KUBE_VERSION/bin/linux/$arch"
       kube_sum_file="$kube_dir/sha256"
 
-      {{- /* create versioned kube dir */}}
+      {{- /* create versioned kube dir */ }}
       mkdir -p "$kube_dir"
       : >"$kube_sum_file"
 
       for bin in kubelet kubeadm kubectl; do
-          {{- /* download kube binary */}}
+          {{- /* download kube binary */ }}
           curl -Lfo "$kube_dir/$bin" "$kube_base_url/$bin"
           chmod +x "$kube_dir/$bin"
 
-          {{- /* download kube binary checksum */}}
+          {{- /* download kube binary checksum */ }}
           sum=$(curl -Lf "$kube_base_url/$bin.sha256")
 
-          {{- /* save kube binary checksum */}}
+          {{- /* save kube binary checksum */ }}
           echo "$sum  $kube_dir/$bin" >>"$kube_sum_file"
       done
 
-      {{- /* check kube binaries checksum */}}
+      {{- /* check kube binaries checksum */ }}
       sha256sum -c "$kube_sum_file"
 
       for bin in kubelet kubeadm kubectl; do
-          {{- /* link kube binaries from verioned dir to $opt_bin */}}
+          {{- /* link kube binaries from verioned dir to $opt_bin */ }}
           ln -sf "$kube_dir/$bin" "$opt_bin"/$bin
       done
 
-      if [[ ! -x /opt/bin/health-monitor.sh ]]; then
-          curl -Lfo /opt/bin/health-monitor.sh https://raw.githubusercontent.com/kubermatic/machine-controller/7967a0af2b75f29ad2ab227eeaa26ea7b0f2fbde/pkg/userdata/scripts/health-monitor.sh
-          chmod +x /opt/bin/health-monitor.sh
-      fi
+  files:
+    - path: /etc/environment
+      content:
+        inline:
+          data: |
+            {{- if .HTTPProxy }}
+            HTTP_PROXY={{ .HTTPProxy }}
+            http_proxy={{ .HTTPProxy }}
+            HTTPS_PROXY={{ .HTTPProxy }}
+            https_proxy={{ .HTTPProxy }}
+            NO_PROXY={{ .NoProxy }}
+            no_proxy={{ .NoProxy }}
+            {{- end }}
 
-    configureProxyScript: |-
-      {{- if .HTTPProxy }}
-      cat <<EOF | tee -a /etc/environment
-      HTTP_PROXY={{ .HTTPProxy }}
-      http_proxy={{ .HTTPProxy }}
-      HTTPS_PROXY={{ .HTTPProxy }}
-      https_proxy={{ .HTTPProxy }}
-      NO_PROXY={{ .NoProxy }}
-      no_proxy={{ .NoProxy }}
-      EOF
-      {{- end }}
+    - path: "/etc/systemd/system/containerd.service.d/environment.conf"
+      content:
+        inline:
+          data: |
+            [Service]
+            Restart=always
+            EnvironmentFile=-/etc/environment
+
+    - path: "/etc/systemd/system/docker.service.d/environment.conf"
+      content:
+        inline:
+          data: |
+            [Service]
+            Restart=always
+            EnvironmentFile=-/etc/environment
```

Comment:
`environment.conf` should only be configured for the respective environment, not unconditionally. That is why a dedicated section(field) `supportedContainerRuntimes` was introduced. It would pick the template for the selected container runtime only https://github.com/kubermatic/operating-system-manager/blob/master/pkg/controllers/osc/resources/operating_system_config.go#L272

Reply:
> `environment.conf` should only be configured for the respective environment, not unconditionally.

That's technically not correct, having both of those files is needed for docker case. For containerd case having `/etc/systemd/system/docker.service.d/environment.conf` makes no difference on the system, except for us so we don't need to repeat our selves. That's why I've rewrote it to "unconditional" state.